package syncro.pb;

option java_package = "uk.me.grambo.syncro.pb";
option optimize_for = LITE_RUNTIME;

message FileHashRequest {
	optional string file_name = 1;
	optional int32 folder_id = 2;
	optional int32 data_size = 3;
	optional string hash = 4;
};

message FileHashResponse {
	required bool ok = 1;
};

message BinaryDataRequest {
	enum TransferDirection {
		Upload = 2;
		Download = 1;
	}
	optional string file_name = 1;
	required int32 folder_id = 2;
	optional int32 recv_buffer_size = 3;
	optional int64 file_size = 4;
	optional TransferDirection direction = 5;
	optional bool one_shot = 6;
	optional int64 start_offset = 7;
};

message BinaryPacketHeader {
	optional int64 file_size = 1;
	optional int64 file_offset = 2;
	optional int32 hash_size = 3;
	
	enum SectionType {
		START = 0;
		MIDDLE = 1;
		END = 2;
	}
	optional SectionType binary_packet_type = 4;
};

message BinaryIncomingResponse {
	optional bool accepted = 1;
	optional int32 max_packet_size = 2;
};

message BinaryIncomingAck {
	optional bool ok = 1;
};
	
